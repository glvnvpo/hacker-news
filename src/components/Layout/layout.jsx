// @flow

import React from 'react';
import {Outlet} from 'react-router-dom';
import './styles.scss';

export const Layout = () => {
	return (
		<>
			<div className='layout bg-orange'>
				<h2 className="color-white">HACKER NEWS</h2>
			</div>
			<Outlet/>
		</>
	);
};