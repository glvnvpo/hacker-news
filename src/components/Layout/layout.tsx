// @flow

import React, {FC} from 'react';
import {Outlet} from 'react-router-dom';
import './styles.scss';

export const Layout: FC = () => {
	return (
		<>
			<div className='layout bg-orange'>
				<h2 className='color-white'>HACKER NEWS</h2>
			</div>
			<Outlet/>
		</>
	);
};