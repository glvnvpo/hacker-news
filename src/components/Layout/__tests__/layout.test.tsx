import React from 'react';
import {mount} from 'enzyme';
import {BrowserRouter, Route, Routes} from "react-router-dom";
import {Layout} from '../layout';

describe('Layout', () => {
	let wrapper;

	it('should render correct layout', () => {
		wrapper = mount(
			<BrowserRouter>
				<Routes>
					<Route element={<Layout/>}>
						<Route path='*' element={<div>test</div>}/>
					</Route>
				</Routes>
			</BrowserRouter>
		);
		expect(wrapper).toMatchSnapshot();
	});
});