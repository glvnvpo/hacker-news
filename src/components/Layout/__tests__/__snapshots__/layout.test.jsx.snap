// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correct layout 1`] = `
<BrowserRouter>
  <Router
    location={
      Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      }
    }
    navigationType="POP"
    navigator={
      Object {
        "action": "POP",
        "createHref": [Function],
        "go": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Routes>
      <RenderedRoute
        match={
          Object {
            "params": Object {
              "*": "",
            },
            "pathname": "/",
            "pathnameBase": "/",
            "route": Object {
              "action": undefined,
              "caseSensitive": undefined,
              "children": Array [
                Object {
                  "action": undefined,
                  "caseSensitive": undefined,
                  "element": <div>
                    test
                  </div>,
                  "errorElement": undefined,
                  "handle": undefined,
                  "hasErrorBoundary": false,
                  "id": "0-0",
                  "index": undefined,
                  "loader": undefined,
                  "path": "*",
                  "shouldRevalidate": undefined,
                },
              ],
              "element": <Layout />,
              "errorElement": undefined,
              "handle": undefined,
              "hasErrorBoundary": false,
              "id": "0",
              "index": undefined,
              "loader": undefined,
              "path": undefined,
              "shouldRevalidate": undefined,
            },
          }
        }
        routeContext={
          Object {
            "matches": Array [
              Object {
                "params": Object {
                  "*": "",
                },
                "pathname": "/",
                "pathnameBase": "/",
                "route": Object {
                  "action": undefined,
                  "caseSensitive": undefined,
                  "children": Array [
                    Object {
                      "action": undefined,
                      "caseSensitive": undefined,
                      "element": <div>
                        test
                      </div>,
                      "errorElement": undefined,
                      "handle": undefined,
                      "hasErrorBoundary": false,
                      "id": "0-0",
                      "index": undefined,
                      "loader": undefined,
                      "path": "*",
                      "shouldRevalidate": undefined,
                    },
                  ],
                  "element": <Layout />,
                  "errorElement": undefined,
                  "handle": undefined,
                  "hasErrorBoundary": false,
                  "id": "0",
                  "index": undefined,
                  "loader": undefined,
                  "path": undefined,
                  "shouldRevalidate": undefined,
                },
              },
            ],
            "outlet": <RenderedRoute
              match={
                Object {
                  "params": Object {
                    "*": "",
                  },
                  "pathname": "/",
                  "pathnameBase": "/",
                  "route": Object {
                    "action": undefined,
                    "caseSensitive": undefined,
                    "element": <div>
                      test
                    </div>,
                    "errorElement": undefined,
                    "handle": undefined,
                    "hasErrorBoundary": false,
                    "id": "0-0",
                    "index": undefined,
                    "loader": undefined,
                    "path": "*",
                    "shouldRevalidate": undefined,
                  },
                }
              }
              routeContext={
                Object {
                  "matches": Array [
                    Object {
                      "params": Object {
                        "*": "",
                      },
                      "pathname": "/",
                      "pathnameBase": "/",
                      "route": Object {
                        "action": undefined,
                        "caseSensitive": undefined,
                        "children": Array [
                          Object {
                            "action": undefined,
                            "caseSensitive": undefined,
                            "element": <div>
                              test
                            </div>,
                            "errorElement": undefined,
                            "handle": undefined,
                            "hasErrorBoundary": false,
                            "id": "0-0",
                            "index": undefined,
                            "loader": undefined,
                            "path": "*",
                            "shouldRevalidate": undefined,
                          },
                        ],
                        "element": <Layout />,
                        "errorElement": undefined,
                        "handle": undefined,
                        "hasErrorBoundary": false,
                        "id": "0",
                        "index": undefined,
                        "loader": undefined,
                        "path": undefined,
                        "shouldRevalidate": undefined,
                      },
                    },
                    Object {
                      "params": Object {
                        "*": "",
                      },
                      "pathname": "/",
                      "pathnameBase": "/",
                      "route": Object {
                        "action": undefined,
                        "caseSensitive": undefined,
                        "element": <div>
                          test
                        </div>,
                        "errorElement": undefined,
                        "handle": undefined,
                        "hasErrorBoundary": false,
                        "id": "0-0",
                        "index": undefined,
                        "loader": undefined,
                        "path": "*",
                        "shouldRevalidate": undefined,
                      },
                    },
                  ],
                  "outlet": null,
                }
              }
            >
              <div>
                test
              </div>
            </RenderedRoute>,
          }
        }
      >
        <Layout>
          <div
            className="layout bg-orange"
          >
            <h2
              className="color-white"
            >
              HACKER NEWS
            </h2>
          </div>
          <Outlet>
            <RenderedRoute
              match={
                Object {
                  "params": Object {
                    "*": "",
                  },
                  "pathname": "/",
                  "pathnameBase": "/",
                  "route": Object {
                    "action": undefined,
                    "caseSensitive": undefined,
                    "element": <div>
                      test
                    </div>,
                    "errorElement": undefined,
                    "handle": undefined,
                    "hasErrorBoundary": false,
                    "id": "0-0",
                    "index": undefined,
                    "loader": undefined,
                    "path": "*",
                    "shouldRevalidate": undefined,
                  },
                }
              }
              routeContext={
                Object {
                  "matches": Array [
                    Object {
                      "params": Object {
                        "*": "",
                      },
                      "pathname": "/",
                      "pathnameBase": "/",
                      "route": Object {
                        "action": undefined,
                        "caseSensitive": undefined,
                        "children": Array [
                          Object {
                            "action": undefined,
                            "caseSensitive": undefined,
                            "element": <div>
                              test
                            </div>,
                            "errorElement": undefined,
                            "handle": undefined,
                            "hasErrorBoundary": false,
                            "id": "0-0",
                            "index": undefined,
                            "loader": undefined,
                            "path": "*",
                            "shouldRevalidate": undefined,
                          },
                        ],
                        "element": <Layout />,
                        "errorElement": undefined,
                        "handle": undefined,
                        "hasErrorBoundary": false,
                        "id": "0",
                        "index": undefined,
                        "loader": undefined,
                        "path": undefined,
                        "shouldRevalidate": undefined,
                      },
                    },
                    Object {
                      "params": Object {
                        "*": "",
                      },
                      "pathname": "/",
                      "pathnameBase": "/",
                      "route": Object {
                        "action": undefined,
                        "caseSensitive": undefined,
                        "element": <div>
                          test
                        </div>,
                        "errorElement": undefined,
                        "handle": undefined,
                        "hasErrorBoundary": false,
                        "id": "0-0",
                        "index": undefined,
                        "loader": undefined,
                        "path": "*",
                        "shouldRevalidate": undefined,
                      },
                    },
                  ],
                  "outlet": null,
                }
              }
            >
              <div>
                test
              </div>
            </RenderedRoute>
          </Outlet>
        </Layout>
      </RenderedRoute>
    </Routes>
  </Router>
</BrowserRouter>
`;
